### PR TITLE
feat(components): add disable animation mixin

### PR DESCRIPTION
**Pull request by tiborux**
EX-6839

### CODE REVIEW

**Review by mnavarroespinosa — DISMISSED** [review]
good job!

File: packages/x-components/src/components/animations/disable-animation.mixin.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  @XInject(DISABLE_ANIMATIONS_KEY)
+  public disableAnimation!: boolean;
+
+  /**
+   * The animation's name based on the {@link disableAnimation} flag.
+   *
+   * @returns The animation name.
+   * @internal
+   */
+  protected get name(): string {
```
Comment (javieri-empathy) [inline packages/x-components/src/components/animations/disable-animation.mixin.ts:34]: Where is the test? :golpederemo:

File: packages/x-components/src/components/animations/disable-animation.mixin.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import { XInject } from '../decorators/injection.decorators';
+import { DISABLE_ANIMATIONS_KEY } from '../decorators/injection.consts';
+
+/**
+ * Mixin to ease disabling animations.
+ *
+ * @public
+ */
+@Component
+export class DisableAnimationMixin extends Vue {
```
Comment (javieri-empathy) [inline packages/x-components/src/components/animations/disable-animation.mixin.ts:12]: Export the mixin

**Review by javieri-empathy — CHANGES_REQUESTED** [review]

**Review by mnavarroespinosa — CHANGES_REQUESTED** [review]
The build seems to be failing because of a JSDoc in `disable-animation.mixin.ts`.

image
 
Maybe you can define it as:

```
  /**
   * The animation's name based on the {@link DisableAnimationMixin.disableAnimation} flag.
   *
   * @returns The animation name.
   * @internal
   */
```

**Review by mnavarroespinosa — APPROVED** [review]

**Review by javieri-empathy — APPROVED** [review]

**Pull request merged**